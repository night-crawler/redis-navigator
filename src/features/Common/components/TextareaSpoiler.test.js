/* eslint-disable react/react-in-jsx-scope,no-undef */
import { TextareaSpoiler } from '.';


describe('TextareaSpoiler', () => {
    it('should render shown TextareaSpoiler', () => {
        const wrapper = shallow(<TextareaSpoiler result='result' show={ true } />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.html()).toMatch(/textarea/i);

        wrapper.find('.hide-shown').simulate('click');
        expect(wrapper.find('.show-hidden')).toHaveLength(1);
    });

    it('should render hidden TextareaSpoiler', () => {
        const wrapper = shallow(<TextareaSpoiler result='result' show={ false } />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.html()).not.toMatch(/textarea/i);

        wrapper.find('.show-hidden').simulate('click');
        expect(wrapper.find('.hide-shown')).toHaveLength(1);
    });
});
