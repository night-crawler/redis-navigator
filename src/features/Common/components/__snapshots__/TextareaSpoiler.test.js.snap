// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextareaSpoiler> should render hidden TextareaSpoiler 1`] = `
<div
  className="TextareaSpoiler"
>
  <div
    className="show-hidden"
    onClick={[Function]}
  >
    <FormattedMessage
      defaultMessage="Show"
      id="features.Common.messages.show"
      values={Object {}}
    />
  </div>
</div>
`;

exports[`<TextareaSpoiler> should render shown TextareaSpoiler 1`] = `
<div
  className="TextareaSpoiler"
>
  <div
    className="hide-shown"
    onClick={[Function]}
  >
    <FormattedMessage
      defaultMessage="Hide"
      id="features.Common.messages.hide"
      values={Object {}}
    />
  </div>
  <styled.textarea
    className="response-data"
    readOnly={true}
    rows={6}
    value="result"
  />
</div>
`;
